### PR TITLE
Switch to `udx-native`

### PR DESCRIPTION
**Pull request by kasperisager**
This PR replaces the networking layer with https://github.com/mafintosh/libudx.

### CODE REVIEW

**Comment by mafintosh** [comment]
Looks good! Keeping this in the PR until we update @hyperswarm/dht to use this.

**Comment by kasperisager** [comment]
I'm working on the Hyperswarm DHT  integration in https://github.com/hyperswarm/dht/pull/82.

**Comment by mafintosh** [comment]
Looks great!

**Comment by mafintosh** [comment]
Once we get some testing milage out of the udx branch on swarm dht, we should land this.

File: lib/io.js [hunk 1]
```diff
@@ -26,14 +27,15 @@ module.exports = class IO {
     this.onresponse = onresponse
     this.ontimeout = ontimeout
 
+    this._udx = new UDX()
```
Comment (mafintosh) [inline lib/io.js:30]: doesn't this need to see if udx is passed in the options?

Comment (kasperisager) [reply lib/io.js:30]: It does indeed!

**Review by kasperisager — COMMENTED** [review]
The test wasn't properly torn down on timeout, causing it to hang.

**Pull request merged**